base/src/test/java/uk/ac/ebi/atlas/experimentimport/analyticsindex/conditions/ConditionsCollectorTest.java

```diff
@@ -0,0 +1,279 @@
+                                            .collect(toSet()))
+                            .addAll(
+                                    sampleCharacteristics.values().stream()
+                                            .flatMap(headerValueMap -> headerValueMap.values().stream())
+                                            .collect(toSet()))
+                            .build());
+    }
+
+    @Test
+    void differentialConditions() {
+        assertThat(subject.getConditions(differentialExperimentMock))
+//                .isEqualTo(legacyConditionsLookupService.buildPropertiesForDifferentialExperiment(differentialExperimentMock.getAccession(), experimentDesignMock, ImmutableList.of(contrast)))
+                .flatExtracting("values")
+                .containsExactlyInAnyOrderElementsOf(
+                        // We compare against a list since EFO terms or factor/sample values might occur more than once
+                        ImmutableList.<String>builder()
+                                .addAll(
+                                        testAssayGroup.assaysAnalyzedForThisDataColumn().stream()
+                                                .flatMap(assayId ->
+                                                        ImmutableSet.<String>builder()
+                                                                .addAll(expandedOntologyTerms.get(assayId))
+                                                                .addAll(factorValues.get(assayId).values())
+                                                                .addAll(sampleCharacteristics.get(assayId).values())
+                                                                .build()
+                                                                .stream())
+                                                .collect(toList()))
+                                .addAll(
+                                        referenceAssayGroup.assaysAnalyzedForThisDataColumn().stream()
+                                                .flatMap(assayId ->
+                                                        ImmutableSet.<String>builder()
+                                                                .addAll(expandedOntologyTerms.get(assayId))
+                                                                .addAll(factorValues.get(assayId).values())
+                                                                .addAll(sampleCharacteristics.get(assayId).values())
+                                                                .build()
+                                                                .stream())
+                                                .collect(toList()))
+                                .build());
+    }
+
+    private static <K, V> Multimap<K, V> randomizedMultimapOf(Collection<K> buckets, Collection<V> values) {
+        checkArgument(values.size() >= buckets.size());
+
+        List<K> _buckets = Lists.newArrayList(buckets);
```

Comment:
What does the underscore in `_variableName` meant to represent...?

Reply:
It’s a way to indicate that it holds a (modified) copy of one of the parameters.